addons/io_hubs_addon/components/utils.py

```diff
@@ -112,3 +121,129 @@ def redraw_component_ui(context):
+
+        try: # Attempt to set up flushing for the C stdout.
+            if platform.system() == "Linux":
+                c_stdout = ctypes.POINTER(_FILE).in_dll(libc, 'stdout')
+            else: # Mac
+                c_stdout = ctypes.POINTER(_FILE).in_dll(libc, '__stdoutp')
+
+            def c_fflush():
+                try:
+                    libc.fflush(c_stdout)
+                except:
+                    print("Warning: Unable to flush the C stdout.")
+
+        except: # The C stdout wasn't found.  This is unlikely to happen, but if it does then just skip flushing since Linux/Mac doesn't seem to strictly require a C-level flush to work.
+            print("Warning: Couldn't get the C stdout.")
+            def c_fflush():
+                pass
+
+    except: # The C runtime wasn't found.  This is unlikely to happen, but if it does then just skip flushing since Linux/Mac doesn't seem to strictly require a C-level flush to work.
+        print("Warning: Unable to find the C runtime.")
+        def c_fflush():
+            pass
+
+
+@contextmanager
+def redirect_c_stdout(binary_stream):
+    stdout_file_descriptor = sys.stdout.fileno()
+    original_stdout_file_descriptor_copy = os.dup(stdout_file_descriptor)
+    pipe_read_end, pipe_write_end = os.pipe() # os.pipe returns two file descriptors.
+
+    try:
+        # Flush the C-level buffer of stdout before redirecting.  This should make sure that only the desired data is captured.
+        c_fflush()
+        # Redirect stdout to your pipe.
+        os.dup2(pipe_write_end, stdout_file_descriptor)
+        yield # wait for input
+    finally:
+        # Flush the C-level buffer of stdout before returning things to normal.  This seems to be mainly needed on Windows because it looks like Windows changes the buffering policy to be fully buffered when redirecting stdout.
+        c_fflush()
+        # Redirect stdout back to the original file descriptor.
+        os.dup2(original_stdout_file_descriptor_copy, stdout_file_descriptor)
+        # Close the write end of the pipe to allow reading.
+        os.close(pipe_write_end)
+        # Read what was written to the pipe and pass it to the binary stream for use outside this function.
+        pipe_reader = os.fdopen(pipe_read_end, 'rb')
+        binary_stream.write(pipe_reader.read())
+        # Close the reader, also closes the pipe_read_end file descriptor.
+        pipe_reader.close()
+        # Close the remaining open file descriptor.
+        os.close(original_stdout_file_descriptor_copy)
+
+def get_host_components(host):
+    for component_item in host.hubs_component_list.items:
+        component_name = component_item.name
+        component_class = get_component_by_name(component_name)
+        if not component_class:
+            continue
+
+        component = getattr(host, component_class.get_id())
+        yield component
+
+def wrap_text(text, max_length=70):
```

Comment:
This is great to have.I think we could extend the use of this to all widgets texts in the future. Blender doesn't wrap by default and some important things like warnings get truncated and are not easily readable.